schemas/manifest.cue

```diff
@@ -53,7 +53,7 @@
     "containerd": {
         "fileName": "moby-containerd_${CONTAINERD_VERSION}+azure-${CONTAINERD_PATCH_VERSION}.deb",
         "downloadLocation": "/opt/containerd/downloads",
-        "downloadURL": "https://moby.blob.core.windows.net/moby/moby-containerd/${CONTAINERD_VERSION}+azure/${UBUNTU_CODENAME}/linux_${CPU_ARCH}/moby-containerd_${CONTAINERD_VERSION}+azure-ubuntu${UBUNTU_RELEASE}u${CONTAINERD_PATCH_VERSION}_${CPU_ARCH}.deb",
+        "downloadURL": "",
```

Comment:
I think it makes more sense to change the install-dependencies.sh script to use this value and have it always set. That way all the info about the containerd version is in the manifest.cue file - rather than having the version in one place and the url in a different place.

Reply:
eg: this PR which uses the URL from manifest.cue for the download location for K8s: https://github.com/Azure/AgentBaker/pull/4443

Reply:
Tim, I am trying to understand what you are suggesting here.
Isn't the install-dependencies.sh script reading the versions and downloadURL? if downloadURL is provided, the script will use it to download the binary. If not, it will then use the version info.
Or do you mean something else?

Reply:
And in general, downloadURL is not needed. That's why I put downloadURL as empty string in manifest.cue. For special case when a particular downloadURL should be used when building a VHD, we will set this manifest.cue.
@anujmaheshwari1 could you confirm if I understand the usage correctly?

Reply:
What I'm really getting at is always store download url in manifest.cue - then we don't need special logic in the install-dependancies.sh script to check and fallback? Having all the data about versions and where to get them from in one place seems quite sensible.

This way we have one code path that we test and verify on each build, rather than have a special case (fallback to the location in the cue file) that's rarely tested.

Reply:
Let me try to understand with an example.
Do you mean changing the manifest.cue to something like the following
```
"containerd": {
   "downloadURL": {
     "1804": <exact URL 1>
     "2004": <exact URL 2>
     "2204": <exact URL 3>
    }
   // remove all other versions.
}
``` 

So that the build pipeline will simply fetch what they need based on the OS SKU and version, instead of having special logic in the install-dependencies.sh to calculate the final URL?

Reply:
No, I mean leaving the download URL as it was in manifest.cue and using it. The vars can be replaced using something like this:

CONTAINERD__URL=$(echo "$CONTAINERD_URL_WITH_VARS" | CONTAINERD_VERSION="${CONTAINERD_VERSION}" CPU_ARCH="${CPU_ARCH}" UBUNTU_CODENAME="${UBUNTU_CODENAME}" CONTAINERD_PATCH_VERSION="${CONTAINERD_PATCH_VERSION}" envsubst)

There's probably a better way to pass vars to envsubst, but that's the way I know :)

Reply:
There might be benefit in your example tho. But I'm not sure we need that additional complexity at this time.

Reply:
One thing worth discussing is that, downloadURL and version are using different ways to get the dependencies.
For example, in [ensureRunc](https://github.com/Azure/AgentBaker/blob/5b923120a139fb3b7b7a6c33a1c1667416ebb72c/parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh#L212):
If it's downloadURL, it uses curl to download the dependency directly.
If it's version, it uses apt-get to install it.
I am not sure without the special logic, how does install-dependencies.sh determine which way to use?

Reply:
the downloadURL that Devin is removing is not used at all during build time. we install containerd from PMC as a package.

unless there is a CVE, for which we need special URL, which can be overwritten in the empty string Devin has currently.

Reply:
Updated the component.cue and component.json based on team's agreement while meeting Tim's requirements